hicommonwealth/commonwealth
# User-Onboarding: Implemented Tags API + Integrated frontend

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/7769

## Description of Changes
- Added `Tags` table to store tags (ex: `Defi`, `DAO`, `Marketplace` etc)
- Added `ProfileTags` table to associate multiple tags to multiple profiles
- Added `CommunityTags` table to associate multiple tags to multiple communities
- Updated edit profile route (`/updateProfile/v2`) to add/remove tags to the user's profile
- Updated the get profile route (`/profile/v2`) to return tags associated with the profile.
- Updated edit community category route (`/updateCommunityCategory`) to add/remove tags to the community
- Updated get communities route (`/communities`) to return tags associated with communities
- Updated edit profile page (`profile/edit`) to use newer tags and allow user to add/remove those tags to/from profile
- Updated edit community profile page (`/:communityId/manage/profile`) to add/remove tags to/from community
- Updated preferences step of welcome onboard modal to allow user to add tags to profile.


https://github.com/hicommonwealth/commonwealth/assets/51641047/2a320096-af7f-4d4a-adb4-3654e44b7ca6



## "How We Fixed It"
N/A

## Test Plan
- Run `yarn migrate-db`
- Add these tags to db
```
INSERT INTO public."Tags" (name, created_at, updated_at) values ('DAO', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('NFTs', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Gaming', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Social', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Memes', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('AI', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('DeFi', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('ReFi', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Lending/Borrowing', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Staking', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Swaps', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('dApp', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Technology', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Security/Auditing', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Governance', now(), now());
INSERT INTO public."Tags" (name, created_at, updated_at) values ('Marketplace', now(), now());
```
- Start the server and log in with a super admin account
- Visit any community profile page `/:communityId/manage/profile` and scroll down to the "Tags" section, verify you see all the newly added tags and are able to add/remove them. Test this process with both navigate-back (submit the form, navigate to any other page, navigate back to this page) and page refresh.
- Visit the edit profile page `profile/edit`, scroll down to the "Preferences" section, and verify you see all the newly added tags and are able to add/remove them. Test this process with both navigate-back (submit the form, navigate to any other page, navigate back to this page) and page refresh.
- Create a new profile, once you see the welcome onboard modal, complete the first step, and then in the "Preferences" step, verify you see all the newly added tags and are able to add/remove them.


## Deployment Plan
N/A

## Other Considerations
The `category` column from the `Communities` table can be deprecated as we now have a newer way to store tags associated to communities. As this pr adds backward compatibility for storing community tags/category, i plan to do the `category` column deprecation work in #7835
